src/material-experimental/mdc-progress-spinner/progress-spinner.ts

```diff
@@ -0,0 +1,262 @@
+
+  /** The diameter of the progress spinner (will set width and height of svg). */
+  @Input()
+  get diameter(): number {
+    return this._diameter;
+  }
+
+  set diameter(size: number) {
+    this._diameter = coerceNumberProperty(size);
+    this._syncFoundation();
+  }
+
+  private _strokeWidth: number;
+
+  /** Stroke width of the progress spinner. */
+  @Input()
+  get strokeWidth(): number {
+    return this._strokeWidth ?? this.diameter / 10;
+  }
+
+  set strokeWidth(value: number) {
+    this._strokeWidth = coerceNumberProperty(value);
+  }
+
+  /** The radius of the spinner, adjusted for stroke width. */
+  _circleRadius(): number {
+    return (this.diameter - BASE_STROKE_WIDTH) / 2;
+  }
+
+  /** The view box of the spinner's svg element. */
+  _viewBox() {
+    const viewBox = this._circleRadius() * 2 + this.strokeWidth;
+    return `0 0 ${viewBox} ${viewBox}`;
+  }
+
+  /** The stroke circumference of the svg circle. */
+  _strokeCircumference(): number {
+    return 2 * Math.PI * this._circleRadius();
+  }
+
+  /** The dash offset of the svg circle. */
+  _strokeDashOffset() {
+    if (this.mode === 'determinate') {
+      return this._strokeCircumference() * (100 - this._value) / 100;
+    }
+    return null;
+  }
+
+  /** Stroke width of the circle in percent. */
+  _circleStrokeWidth() {
+    return this.strokeWidth / this.diameter * 100;
+  }
+
+  ngAfterViewInit() {
+    this._foundation = new MDCCircularProgressFoundation(this._adapter);
+    this._foundation.init();
+    this._syncFoundation();
+  }
+
+  ngOnDestroy() {
+    if (this._foundation) {
+      this._foundation.destroy();
+    }
+  }
+
+  private _syncFoundation() {
+    const foundation = this._foundation;
+
+    if (foundation) {
+      const mode = this.mode;
+      foundation.setDeterminate(mode === 'determinate');
+      foundation.setProgress(this.value / 100);
+    }
+  }
+}
+
+/**
+ * `<mat-spinner>` component.
+ *
+ * This is a component definition to be used as a convenience reference to create an
+ * indeterminate `<mat-progress-spinner>` instance.
+ */
+@Component({
+  selector: 'mat-spinner',
+  host: {
+    'role': 'progressbar',
+    'mode': 'indeterminate',
+    'class': 'mat-mdc-spinner mat-mdc-progress-spinner',
+    '[class._mat-animation-noopable]': `_noopAnimations`,
+    '[style.width.px]': 'diameter',
+    '[style.height.px]': 'diameter',
+  },
+  inputs: ['color'],
+  templateUrl: 'progress-spinner.html',
+  styleUrls: ['progress-spinner.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MatSpinner extends MatProgressSpinner {
+  constructor(elementRef: ElementRef<HTMLElement>,
+              @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode: string,
+              @Inject(MAT_PROGRESS_SPINNER_DEFAULT_OPTIONS)
+                defaults?: MatProgressSpinnerDefaultOptions) {
+    super(elementRef, animationMode, defaults);
+    this.mode = 'indeterminate';
+  }
+}
```

Comment:
In hindsight, having a separate component for `MatSpinner` is probably generating more code than is strictly necessary. Seems like we could omit this class completely, add `mat-spinner` to the other component's selector, and change the `mode` property initialization to
```ts
mode = this.elementRef.nativeElement.nodeName.toLowerCase() === 'mat-spinner' ? 
    'indeterminate' : 
    'determinate';
```
and then be backwards compatible by adding this to the end of the file
```ts
export {MatProgressSpinner as MatSpinner};
```